### PR TITLE
Fixed maxConcurrency issue in HTTP channel

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
In case of error, the item was not removed from the channel

### CODE REVIEW

File: pkg/channel/http/http_channel.go [hunk 1]
```diff
@@ -167,8 +167,13 @@ func (h *Channel) invokeMethodV1(ctx context.Context, req *invokev1.InvokeMethod
...
 
 	if h.ch != nil {
-		h.ch <- 1
+		h.ch <- struct{}{}
 	}
+	defer func() {
+		if h.ch != nil {
+			<-h.ch
+		}
+	}()
```
Comment (halspang) [inline pkg/channel/http/http_channel.go:176]: Seems like we should get a test for this. Is that possible?

**Review by halspang — COMMENTED** [review]
Change seems fine, just want a test for it if possible.

File: pkg/channel/http/http_channel.go (hunk 1 above)
Comment (ItalyPaleAle) [reply pkg/channel/http/http_channel.go:176]: Added test

**Comment by ItalyPaleAle** [comment]
@halspang I added the tests, can this be approved and merged now?

**Review by daixiang0 — APPROVED** [review]

**Pull request merged**